Spigot-Server-Patches/0548-Don-t-mark-null-chunk-sections-for-block-updates.patch

```diff
@@ -0,0 +1,18 @@
+From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
+From: Mariell Hoversholm <proximyst@proximyst.com>
+Date: Fri, 14 Aug 2020 23:41:19 +0200
+Subject: [PATCH] Don't mark null chunk sections for block updates
+
+
+diff --git a/src/main/java/net/minecraft/server/PlayerChunk.java b/src/main/java/net/minecraft/server/PlayerChunk.java
+index 92a5a5cefb048d4a65ba64e154ce2a3459b28b28..e02ce1c93a6b0a1789212a60a789f7dada82a0fe 100644
+--- a/src/main/java/net/minecraft/server/PlayerChunk.java
++++ b/src/main/java/net/minecraft/server/PlayerChunk.java
+@@ -363,6 +363,7 @@ public class PlayerChunk {
+ 
+         if (chunk != null) {
+             byte b0 = (byte) SectionPosition.a(blockposition.getY());
++            if (chunk.getSections()[b0] == null) return; // Paper - Don't mark null chunk sections for block updates
```

Comment:
I was working on a similar problem and think this solution may cause issues. I believe it's still valid to try to update positions in null chunk sections for purposes of updating fake blocks. I opened a related issue to spigot: https://hub.spigotmc.org/jira/projects/SPIGOT/issues/SPIGOT-6076

Reply:
In that case, we have to make all blocks in that section air before it starts accessing the section. Does this sound sane?

Reply:
I was hoping the fix here would be to init and add a ChunkSection if it's null, IE `chunk.getSections()[b0] = new ChunkSection(blockposition.getY() >> 4 << 4);`. Don't have any experience patching the server so I haven't been able to see if that works. Is that what you mean?

Also, this might be better deeper down where the server handles building its PacketPlayOutMultiBlockChange packet in PlayerChunk. Single block updates in don't cause any errors/crashes since they don't reference the chunk section

Reply:
I've done this in the latest commit.